kf/kf_unix.go

```diff
@@ -115,6 +115,43 @@ func VerifyNMountBPFFS() error {
 			return fmt.Errorf("unable to mount %s at %s: %s", srcPath, dstPath, err)
 		}
 	}
+
+	return VerifyNMountTraceFS()
+}
+
+// VerifyNMounTraceFS - Mounting trace filesystem
+func VerifyNMountTraceFS() error {
+	dstPath := "/sys/kernel/tracing"
+	srcPath := "tracefs"
+	fstype := "tracefs"
+	flags := syscall.MS_NODEV | syscall.MS_NOEXEC | syscall.MS_NOSUID | syscall.MS_RELATIME
+
+	mnts, err := ioutil.ReadFile("/proc/mounts")
```

Comment:
Is there a golang equivalent of https://man7.org/linux/man-pages/man3/getmntent.3.html ? I would prefer not to use a proc based API because what if proc itself is not mounted (yeah it happens).
Also, assuming there is no such equivalent, I am not sure if all users can open /proc/mounts (I think it is restricted to CAP_SYS_ADMIN -- please confirm independently). If the access is restricted, then I would suggest using /proc/self/mounts might be the right way forward.

Reply:
I would think this is pretty uncommon case of not having /proc and it does seem like /proc/self/mountinfo is the best place to read from and doesn't seem to require CAP_SYS_ADMIN